File: search_ua.py

with open(r"c:/Users/Will/Downloads/Teknov8/Teknov8/node_modules/vercel/dist/index.js", "r", encoding="utf-8") as f:
    content = f.read()
    index = content.find("User-Agent")
    if index != -1:
        print(f"Found at {index}")
        print(content[index-100:index+200])
    else:
        print("Not found 'User-Agent'")

    # Try lowercase
    index = content.find("user-agent")
    if index != -1:
        print(f"Found lowercase at {index}")
        print(content[index-100:index+200])
    
    # Try searching for the construction pattern
    # "vercel " + version
    index = content.find("vercel ")
    if index != -1:
        print(f"Found 'vercel ' at {index}")
        print(content[index:index+100])

File: search_ua.js

const fs = require('fs');
const path = 'node_modules/vercel/dist/index.js';
const content = fs.readFileSync(path, 'utf8');

let index = content.indexOf('User-Agent');
if (index !== -1) {
    console.log(`Found 'User-Agent' at ${index}`);
    console.log(content.substring(index - 100, index + 200));
} else {
    console.log("Not found 'User-Agent'");
}

index = content.indexOf('user-agent');
if (index !== -1) {
    console.log(`Found 'user-agent' at ${index}`);
    console.log(content.substring(index - 100, index + 200));
}

// Search for the specific error string part
index = content.indexOf('win32 (x64)');
if (index !== -1) {
    console.log(`Found 'win32 (x64)' at ${index}`);
    console.log(content.substring(index - 100, index + 200));
}
